prime-router/src/main/kotlin/cli/SenderUtilsCommand.kt

```diff
@@ -0,0 +1,158 @@
+    settingType = SettingType.SENDER,
+    operation = Operation.GET
+) {
+    val publicKeyFilename by option("--public-key",
+        help = "Path to public key .pem file",
+        metavar = "<public-pem-key-file>")
+        .required()
+
+    private val kid by option(
+        "--kid",
+        metavar = "<string key id>",
+        help = """
+            Specify desired key id for this key.
+            When sender makes a token req, the kid in jwt must match this kid.
+            If not set, use the sender fullname as the kid value
+        """.trimIndent()
+    )
+
+    private val scope by option(
+        "--scope",
+        metavar = "<desired scope>",
+        help = "Specify desired authorization scope.  Example:  'report' to request access to the 'report' endpoint."
+    ).required()
+
+    val doIt by option(
+        "--doit",
+        help = "Save the modified Sender setting to the database (default is to just print the modified setting)"
+    ).flag(default = false)
+
+    override fun run() {
+        val environment = getEnvironment()
+        val accessToken = getAccessToken(environment)
+        val publicKeyFile = File(publicKeyFilename)
+        if (! publicKeyFile.exists()) {
+            echo("Unable to fine pem file " + publicKeyFile.absolutePath)
+            return
+        }
+        if (!Scope.isWellFormedScope(scope)) {
+            echo("$scope is not a well formed scope value")
+            return
+        }
+        val jwk = SenderUtils.readPublicKeyPemFile(publicKeyFile)
+        jwk.kid = if (kid.isNullOrEmpty()) settingName else kid
+
+        val origSenderJson = get(environment, accessToken, settingType, settingName)
+        val origSender = Sender(jsonMapper.readValue(origSenderJson, SenderAPI::class.java))
+
+        if (!Scope.isValidScope(scope, origSender)) {
+            echo("Sender full name in scope must match $settingName.  Instead got: $scope")
+            return
+        }
+
+        val newSender = Sender(origSender, scope, jwk)
+        val newSenderJson = jsonMapper.writeValueAsString(newSender)
+
+        echo("*** Original Sender *** ")
+        if (useJson) writeOutput(origSenderJson) else writeOutput(toYaml(origSenderJson, settingType))
+        echo("*** End Original Sender *** ")
+        echo("*** Modified Sender, including new key *** ")
+        if (useJson) writeOutput(newSenderJson) else writeOutput(toYaml(newSenderJson, settingType))
+        echo("*** End Modified Sender, including new key *** ")
+        if (!doIt) {
+            echo("""
+
+            Nothing has been updated or changed. 
+            To add the key permanently, review, then rerun this same command including the --doit option
+        """.trimIndent())
+            return
+        }
+        val response = put(environment, accessToken, settingType, settingName,newSenderJson)
+        echo()
+        echo(response)
+        echo()
+    }
+
+}
+
+class TokenUrl : SingleSettingCommand (
```

Comment:
New helper cli to simulate what a sender might do.    No, we are NOT intending that our senders use our CLI.  This was very useful in testing and demo-ing, however.